src/plugins/data/common/index_patterns/types.ts

```diff
@@ -26,19 +26,32 @@ import { KBN_FIELD_TYPES, IndexPatternField, FieldFormat } from '..';
 
 export type FieldFormatMap = Record<string, SerializedFieldFormat>;
 
+/**
+ * IIndexPattern allows for an IndexPattern OR an index pattern saved object
```

Comment:
Not sure I understand this comment, especially as `IIndexPattern` is used extensively ATM